admin/app/tools/DfpLink.scala

```diff
@@ -24,21 +25,25 @@ object DfpLink {
 
 object SiteLink {
 
-  def adUnit(path: String): Option[String] = {
+  def adUnit(path: String, adTest: Option[String]): String = {
 
     def getRelativePath(dropFromRight: Int) = path.split("/").drop(1).dropRight(dropFromRight).mkString("/")
 
-    lazy val relativePath = getRelativePath(1)
-    lazy val relativeEditionalisedPath = getRelativePath(2)
-
-    if (path endsWith "/front") {
-      Some(s"${site.host}/${relativePath}")
-    } else if (path endsWith "/front/ng") {
-      Some(s"${site.host}/${relativeEditionalisedPath}?view=mobile")
-    } else if (path endsWith "/front/r2") {
-      Some(s"${site.host}/${relativeEditionalisedPath}?view=classic")
-    } else {
-      None
+    lazy val domain = if (adTest.isDefined) commercial.testDomain else site.host
+
+    val relativePath =
+      if (path endsWith "/front/ng")
+        getRelativePath(2)
+      else if (path endsWith "front")
+        getRelativePath(1)
+      else
+        getRelativePath(0)
```

Comment:
This should never happen because pageskins are only allowed on fronts.  It might be worth adding a test to make sure this is the case.